codec-http/src/main/java/io/netty/handler/codec/http/multipart/HttpPostBodyUtil.java

```diff
@@ -172,6 +172,34 @@ static int findLineBreak(ByteBuf buffer, int index) {
         return posFirstChar;
     }
 
+    /**
+     * Try to find last LF or CRLF as Line Breaking
+     *
+     * @param buffer the buffer to search in
+     * @param index the index to start from in the buffer
+     * @return a relative position from index > 0 if LF or CRLF is found
+     *         or < 0 if not found
+     */
+    static int findLastLineBreak(ByteBuf buffer, int index) {
+        int toRead = buffer.readableBytes() - (index - buffer.readerIndex());
+        int posFirstChar = buffer.bytesBefore(index, toRead, HttpConstants.LF);
+        if (posFirstChar == -1) {
+            // No LF, so neither CRLF
+            return  -1;
+        }
+        int foundLF = 1;
+        if (posFirstChar > 0 && buffer.getByte(index + posFirstChar - 1) == HttpConstants.CR) {
+            posFirstChar--;
+            foundLF++;
+        }
+        // Might be not the last LF or CRLF
+        int nextPos = findLastLineBreak(buffer, index + posFirstChar + foundLF);
```

Comment:
I'd prefer this method was iterative rather than recursive. Recursion, especially when based on external input, can cause unpredictable stack usage. It's also harder on the JIT.

Reply:
OK, let me try to break the recursion.

Reply:
Done